backend/src/lib.rs

```diff
@@ -434,8 +434,16 @@ impl LTN {
 
     #[wasm_bindgen(js_name = getShortcutsCrossingRoad)]
     pub fn get_shortcuts_crossing_road(&self, road: usize) -> Result<String, JsValue> {
+        let derived = self
```

Comment:
These two calls were wastefully recalculating shortcuts from scratch, even though we already calculate and store them. `derived` should never be `None` past the constructor, and every mutation calls `after_edit`, so this is safe